src/enumerable.cr

```diff
@@ -303,8 +303,16 @@ module Enumerable(T)
     raise ArgumentError.new("size must be positive") if size <= 0
 
     each_slice(size) do |slice|
-      group = slice.size < size ? slice + [filled_up_with] : slice
-      yield group
+      if slice.size < size
+        paddings = Array(T | U).new
+        (size - slice.size).times do
+          paddings << filled_up_with
+        end
+        group = slice + paddings
+        yield group
+      else
+        yield slice
+      end
```

Comment:
Thank you! The code looks simple to me.
But I think the `slice` is declared to `Array(T)` at https://github.com/crystal-lang/crystal/blob/0cc341dfc0fdc428b641356f3bd10b7a7339a514/src/enumerable.cr#L160.
Can I append the nil or given value to the slice at here?

I have tried some code as below they show errors.

```crystal
group = slice.as(Array(T | U)) # => can't cast Array(Int32) to Array(Int32 | Nil)
```